### DISCUSSION TITLE
Velocity integration reaching a max?

### DISCUSSION THREAD

**Discussion by jebremus** [bepu/bepuphysics2#116] [Q&A]
Hello!

I'm simulating simple gravity just like it's done in DemoCallbacks' IntegrateVelocity, and I'm seeing falling objects quickly hit terminal velocity - their acceleration downward spikes, then slows until it hits zero and the object continues falling at the same speed forever.  I'm not finding any velocity maximums or explicit terminal velocity code anywhere; is this something anyone else has asked about? Thanks much!

**Comment by RossNordby** [comment]
The DemoPoseIntegratorCallbacks include a damping term: https://github.com/bepu/bepuphysics2/blob/master/Demos/DemoCallbacks.cs#L52

That saps a fraction of the velocity out of a body over time, so acceleration gradually declines to zero as the acceleration of gravity is counteracted by the deceleration of damping.

Those callbacks allow you set the damping fraction in the constructor. Removing the term entirely or setting damping to zero will let objects accelerate forever (well, until numerical issues start happening).

**Reply by jebremus** [reply]
Oh perfect, thanks. I just totally misread the damping bit. Thanks again!
